internal/utils/utils.go

```diff
@@ -123,91 +124,9 @@ func IsEmpty(v interface{}) bool {
-					Optional:     true,
-					Sensitive:    true,
-					RequiredWith: []string{"elasticsearch_connection.0.username"},
-				},
-				"api_key": {
-					Description:   "API Key to use for authentication to Elasticsearch",
-					Type:          schema.TypeString,
-					Optional:      true,
-					Sensitive:     true,
-					ConflictsWith: []string{"elasticsearch_connection.0.username", "elasticsearch_connection.0.password"},
-				},
-				"endpoints": {
-					Description: "A list of endpoints the Terraform provider will point to. They must include the http(s) schema and port number.",
-					Type:        schema.TypeList,
-					Optional:    true,
-					Sensitive:   true,
-					Elem: &schema.Schema{
-						Type: schema.TypeString,
-					},
-				},
-				"insecure": {
-					Description: "Disable TLS certificate validation",
-					Type:        schema.TypeBool,
-					Optional:    true,
-					Default:     false,
-				},
-				"ca_file": {
-					Description:   "Path to a custom Certificate Authority certificate",
-					Type:          schema.TypeString,
-					Optional:      true,
-					ConflictsWith: []string{"elasticsearch_connection.0.ca_data"},
-				},
-				"ca_data": {
-					Description:   "PEM-encoded custom Certificate Authority certificate",
-					Type:          schema.TypeString,
-					Optional:      true,
-					ConflictsWith: []string{"elasticsearch_connection.0.ca_file"},
-				},
-				"cert_file": {
-					Description:   "Path to a file containing the PEM encoded certificate for client auth",
-					Type:          schema.TypeString,
-					Optional:      true,
-					RequiredWith:  []string{"elasticsearch_connection.0.key_file"},
-					ConflictsWith: []string{"elasticsearch_connection.0.cert_data", "elasticsearch_connection.0.key_data"},
-				},
-				"key_file": {
-					Description:   "Path to a file containing the PEM encoded private key for client auth",
-					Type:          schema.TypeString,
-					Optional:      true,
-					RequiredWith:  []string{"elasticsearch_connection.0.cert_file"},
-					ConflictsWith: []string{"elasticsearch_connection.0.cert_data", "elasticsearch_connection.0.key_data"},
-				},
-				"cert_data": {
-					Description:   "PEM encoded certificate for client auth",
-					Type:          schema.TypeString,
-					Optional:      true,
-					RequiredWith:  []string{"elasticsearch_connection.0.key_data"},
-					ConflictsWith: []string{"elasticsearch_connection.0.cert_file", "elasticsearch_connection.0.key_file"},
-				},
-				"key_data": {
-					Description:   "PEM encoded private key for client auth",
-					Type:          schema.TypeString,
-					Optional:      true,
-					RequiredWith:  []string{"elasticsearch_connection.0.cert_data"},
-					ConflictsWith: []string{"elasticsearch_connection.0.cert_file", "elasticsearch_connection.0.key_file"},
-				},
-			},
-		},
-	}
+	connectionKeyName := "elasticsearch_connection"
+
+	providedSchema[connectionKeyName] = providerSchema.GetConnectionSchema(connectionKeyName)
```

Comment:
Would probably move this to a constant.

```suggestion
const resourceEsConnectionSchemaKey = "elasticsearch_connection"

// Returns the common connection schema for all the Elasticsearch resources,
// which defines the fields which can be used to configure the API access
func AddConnectionSchema(providedSchema map[string]*schema.Schema) {
	providedSchema[resourceEsConnectionSchemaKey] = providerSchema.GetConnectionSchema(resourceEsConnectionSchemaKey)
```